### ISSUE TITLE
global does not seem to be considered when matching for transform

### ISSUE THREAD

**Issue by MartinKl** [grew-nlp/grew#17]
I prepared a minimal example to illustrate the problem. My input data:

```
# text = Gefahrene
1	fahren	fahren	V-PPast-ADJ-SUFF-SUFF-NN	_	_	_	_	_	_

# text = Verletzter
1	verletzen	verletzen	V-PPast-ADJ-SUFF-SUFF-NN	_	_	_	_	_	_
```

A line in a conll file here represents a morpheme or allomorph, more or less, the upos tag represents the  morpheme's category. The sentence holds the sequence of all morphemes of a word which is the outcome of all applied processes (dependency relations) and coded in the metadata field `text`.

The two cases above should be treated differently, which I'd like to do by looking at the metadata of a word/sentence via `global`:

```
package minimal_set_of_rules {
	rule a { % deverbal that operates on ge-past form, so (ge-(verb))-e
		global { text = re"G.*" }
		pattern { B [upos=V-PPast-ADJ-SUFF-SUFF-NN]; }		
		commands {
			add_node G :< B;
			G.form=Ge;
			G.upos=V-PPast;
			B.upos=V;
			B.postop="conv:A,trans:N";
			add_edge B -[der:V]-> G;
		}		
	}
	rule b { % deverbal that operates on ge-past form, but ge- is dropped bc the input verb is a prefixverb		
		global { text = re"[^G].*" }
		pattern { B [upos=V-PPast-ADJ-SUFF-SUFF-NN]; }
		commands {			
			B.upos=V;
			B.postop="conv:A,trans:N";
		}
	}
}

strat main {
	Onf(minimal_set_of_rules)
}
```

The second rule is supposed to treat the second example differently such that no prefix node "Ge" is introduced by the very general heuristic "does not start with a G". The desired outcome is:

```
# text = Gefahrene
1	Ge	_	V-PPast	_	_	2	der:V	_	_
2	fahren	fahren	V	V	_	_	_	_	postop=conv:A,trans:N

# text = Verletzter
1	verletzen	verletzen	V	_	_	_	_	_	postop=conv:A,trans:N
```

The created output, though, seems to imply that the `global` filter does not work properly and rule b is applied to both examples:

```
# text = Gefahrene
1	fahren	fahren	V	_	_	_	_	_	postop=conv:A,trans:N

# text = Verletzter
1	verletzen	verletzen	V	_	_	_	_	_	postop=conv:A,trans:N
```

I am using Grew 1.9.0 with libgrew 1.9.1 on Ubuntu 20.04.

My suspicion is that "global" is not evaluated when matching for transform. Thanks for the support!

**Comment by bguil** [comment]
This should be fixed by [29b8ec06](https://gitlab.inria.fr/grew/libcaml-grew/-/commit/29b8ec06ff3a8ca400e770e19cdd3d66b4661d5e).

A new version 1.9.2 of `libcaml-grew` is available. To update, run 

```
opam update
opam upgrade
```

**Comment by MartinKl** [comment]
Great, thanks! Works as expected:

```
# global.columns = ID FORM LEMMA UPOS XPOS FEATS HEAD DEPREL DEPS MISC
# text = Gefahrene
1	Ge	_	V-PPast	_	_	2	der:V	_	preop=conv:A
2	fahren	fahren	V	_	_	_	_	_	postop=conv:A,trans:N

# text = Verletzter
1	verletzen	verletzen	V	_	_	_	_	_	postop=conv:A,trans:N
```
